cmd/podman/images/scp.go

```diff
@@ -0,0 +1,224 @@
+		if terminal.IsTerminal(int(fi.Fd())) {
+			return errors.Errorf("refusing to save to terminal. Use -o flag or redirect")
+		}
+		pipePath, cleanup, err := setupPipe()
+		if err != nil {
+			return err
+		}
+		if cleanup != nil {
+			defer func() {
+				errc := cleanup()
+				if succeeded {
+					writeErr := <-errc
+					if writeErr != nil && finalErr == nil {
+						finalErr = writeErr
+					}
+				}
+			}()
+		}
+		scpOpts.Save.Output = pipePath
+
+		scpOpts.Load.Input = scpOpts.Save.Output
+	}
+	if err := parse.ValidateFileName(saveOpts.Output); err != nil {
+		return err
+	}
+	confR, err := config.NewConfig("")
+	if err != nil {
+		return errors.Wrapf(err, "could not make config")
+	}
+	abiEng, err := registry.NewImageEngine(cmd, args)
+	if err != nil {
+		return errors.Wrapf(err, "could not make abi engine")
+	}
+	confR.Engine = config.EngineConfig{Remote: true, CgroupManager: "cgroupfs"}
+
+	iden := cmd.Flags().Lookup("identity").Value.String()
+
+	url := cmd.Flags().Lookup("url").Value.String()
+	optionsTunnel := entities.PodmanConfig{Config: confR, EngineMode: entities.TunnelMode, Identity: iden, URI: url}
+
+	conn := cmd.Flags().Lookup("connection")
+	if conn != nil && conn.Changed {
+		optionsTunnel.Engine.ActiveService = conn.Value.String()
+		var err error
+		optionsTunnel.URI, optionsTunnel.Identity, err = optionsTunnel.Config.ActiveDestination()
+		if err != nil {
+			return errors.Wrap(err, "failed to resolve active destination")
+		}
+		if err := cmd.Flags().Set("url", optionsTunnel.URI); err != nil {
+			return errors.Wrap(err, "failed to override --url flag")
+		}
+		if err := cmd.Flags().Set("identity", optionsTunnel.Identity); err != nil {
+			return errors.Wrap(err, "failed to override --identity flag")
+		}
+	}
+	tunEng, err := infra.NewImageEngine(&optionsTunnel)
+	if err != nil {
+		return (errors.Wrapf(err, "could not make tunnel engine"))
+	}
+	if err != nil {
+		return (errors.Wrapf(err, "could not make abi engine"))
+	}
+	abiErr := abiEng.Save(context.Background(), args[0], tags, scpOpts.Save)
+	if abiErr != nil {
+		errors.Wrapf(abiErr, "could not save image as specified")
+	}
+	tunOut, err := tunEng.Load(context.Background(), scpOpts.Load)
+	if err != nil {
+		errors.Wrapf(err, "could not load saved image")
+	}
+	if tunOut != nil {
+		out := tunOut.Names
+		fmt.Println("Loaded image(s): " + strings.Join(out, ","))
+	}
+	return nil
+}
+
+func resolveDestination() (string, string, string) {
+	if uri, found := os.LookupEnv("CONTAINER_HOST"); found {
+		var ident string
+		if v, found := os.LookupEnv("CONTAINER_SSHKEY"); found {
+			ident = v
+		}
+		return "", uri, ident
+	}
+	cfg, err := config.ReadCustomConfig()
+	if err != nil {
+		logrus.Warning(errors.Wrap(err, "unable to read local containers.conf"))
+		return "", registry.DefaultAPIAddress(), ""
+	}
+	uri, ident, err := cfg.ActiveDestination()
+	if err != nil {
+		return "", registry.DefaultAPIAddress(), ""
+	}
+	return cfg.Engine.ActiveService, uri, ident
+}
```

Comment:
Please add comments to the code.

I guarantee that my brain will garbage collect the design decisions here in the next couple of weeks :^)

Reply:
Will do, just changed a lot of the code!